### PR TITLE
feat(core): add oakterm-mux split-tree layout model (TREK-96)

### PR DESCRIPTION
**Pull request by jbabin91**
## Summary

Introduces the **`oakterm-mux`** crate housing the multiplexer domain model, starting with the n-ary split-tree layout — the foundation for EPIC-8 (Tiled Splits) and the rest of Phase 1. Implements the Contract + Auto-Flattening sections of **Spec-0007**, per **ADR-0010** (n-ary tree with auto-flattening).

Scope is intentionally limited to the data structure: types, the `flatten` auto-flattening operation, invariant validation, and tests. Split/close/resize tree operations are **TREK-97**.

## What's here

- **Types** — `LayoutNode` (`Container | Leaf`), `Container { direction, children: Vec<Child> }`, `Child { node, weight }`, `SplitDirection`, `PaneId`.
- **`flatten`** — restores structural canonical form bottom-up: single-child containers collapse into their sole child (which inherits the container's weight), and same-direction nested containers merge in with weights scaled by the child's weight. Idempotent and **total** (never panics).
- **`validate` / `is_canonical`** — enforce every Container invariant: ≥2 children, positive + finite weights, sum-to-1.0 within `WEIGHT_SUM_TOLERANCE`, no same-direction nesting.
- **30 unit tests** covering the flatten algebra (collapse, merge, weight scaling, deep nesting, idempotence, per-subtree canonicalization), all validation edge cases (NaN/±inf/negative/zero weights, tolerance boundary, recursion), and pane-id ordering.

## Design notes

- **`Vec<Child>` over parallel `children`/`weights` Vecs** — pairing the weight with the node makes a length mismatch *unrepresentable*, which is why `flatten` is total and there's no runtime length check.
- **No serde on the in-memory model** — the wire/session JSON is owned by Spec-0010's separate `SavedLayoutNode` DTOs (EPIC-12), which convert to/from this tree; deriving serde here would misrepresent that contract.

## Testing

- `cargo test -p oakterm-mux` — 30 passing
- `mise run test` — full workspace suite green (0 failures)
- `mise run check` — clippy (pedantic), fmt, and markdown lint all clean

## Review

Went through the full `/review-cycle` (Codex + code-reviewer + silent-failure-hunter + type-design + spec-conformance + maintainability). The `Vec<Child>` representation and serde deferral were both applied from review findings. Spec-conformance: faithful, no gaps.

Also bumps Spec-0007 `accepted → implementing`.

TREK-96

https://claude.ai/code/session_01UrFUGdnbeMHUN2FPvZJnuC

### CODE REVIEW

File: crates/oakterm-mux/src/layout.rs [hunk 1]
```diff
@@ -0,0 +1,636 @@
...
+    fn container(
+        direction: SplitDirection,
+        nodes: Vec<LayoutNode>,
+        weights: Vec<f32>,
+    ) -> LayoutNode {
+        let children = nodes
+            .into_iter()
+            .zip(weights)
+            .map(|(node, weight)| Child::new(node, weight))
+            .collect();
```
[the remark this answers is not in the document]
Comment (jbabin91) [reply crates/oakterm-mux/src/layout.rs:282]: Good catch — fixed in ae89e77. The `container` test helper now asserts `nodes.len() == weights.len()` before zipping, so a mismatched test setup fails loudly instead of silently truncating.
